### DISCUSSION TITLE
Zarr Python Core Devs: Please Review the V3 Design Doc

### DISCUSSION THREAD

**Discussion by rabernat** [zarr-developers/zarr-python#1591] [General]
This is an announcement aimed at the @zarr-developers/python-core-devs. 

There is some major action happening in zarr-python right now! A group of motivated developers are at work refactoring the package with the following goals:

- Provide a complete implementation of Zarr V3 through the Zarr-Python API
- Align the Zarr-Python array API with the [array API Standard](https://data-apis.org/array-api/latest/)
- Clear the way for exciting extensions / ZEPs (i.e. [sharding](https://zarr-specs.readthedocs.io/en/latest/v3/codecs/sharding-indexed/v1.0.html), [variable chunking](https://zarr.dev/zeps/draft/ZEP0003.html), etc.)
- Provide a developer API that can be used to implement and register V3 extensions
- Improve the performance of Zarr-Python by streamlining the interface between the Store layer and higher level APIs (e.g. Groups and Arrays)
- Clean up the internal and user facing APIs 
- Improve code quality and robustness (e.g. achieve 100% type hint coverage)

The outcome will be a new major version release of zarr-python (3.0).

This is being spearheaded by @jhamman, @d-v-b, and @normanrz. 🙌 Currently these changes are being made on the [v3 branch](https://github.com/zarr-developers/zarr-python/tree/v3).  However, getting such a major change over the finish line and into `main` will require the involvement of many more of you.

To prepare for this, I'd like to ask two things from everyone on @zarr-developers/python-core-devs:
1. Decide if you want to remain as a core dev. Many of you have become inactive. That's a normal part of the lifecycle of any open source project. If you don't plan on becoming active again, please let us know in the thread below.
2. For those of you who plan to remain active, please review the design doc at https://github.com/zarr-developers/zarr-python/pull/1583 and give your feedback.

Going forward, there will be quite a few PRs to review in the coming weeks (see, for example #1589 and #1590). We're going to need all hands on deck!

Thanks for your contributions to Zarr!

**Comment by jeromekelleher** [comment]
As an interested user I'd like to help out a bit with these changes @rabernat, but I'm certainly not a core-dev. Please remove me from the team, and thanks for all the awesome work!

**Reply by jhamman** [reply]
Thanks @jeromekelleher!

**Comment by shoyer** [comment]
I am not an active core developer, so feel free to remove me!

**Reply by jhamman** [reply]
Thanks @shoyer!
